jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/SetPublicSubjectIdentifierPerClient.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Janssen Project software is available under the Apache License (2004). See http://www.apache.org/licenses/ for full text.
+ *
+ * Copyright (c) 2020, Janssen Project
+ */
+
+package io.jans.as.client.ws.rs;
+
+import io.jans.as.client.*;
+import io.jans.as.model.common.ResponseType;
+import io.jans.as.model.common.SubjectType;
+import io.jans.as.model.crypto.signature.RSAPublicKey;
+import io.jans.as.model.crypto.signature.SignatureAlgorithm;
+import io.jans.as.model.exception.InvalidJwtException;
+import io.jans.as.model.jws.RSASigner;
+import io.jans.as.model.jwt.Jwt;
+import io.jans.as.model.jwt.JwtClaimName;
+import io.jans.as.model.jwt.JwtHeaderName;
+import io.jans.as.model.register.ApplicationType;
+import io.jans.as.model.util.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.testng.ITestContext;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.testng.Assert.*;
+
+/**
+ * @author Javier Rojas Blum
+ * @version February 10, 2022
+ */
+public class SetPublicSubjectIdentifierPerClient extends BaseTest {
+
+    @Test(dataProvider = "dataMethod")
+    public void setPublicSubjectIdentifierPerClient(
+            final String redirectUri, final String userId, final String userSecret,
+            final String subjectIdentifierAttribute, final String expectedSubValue) throws InvalidJwtException {
+        showTitle("setPublicSubjectIdentifierPerClient");
+
+        List<ResponseType> responseTypes = Arrays.asList(
+                ResponseType.TOKEN,
+                ResponseType.ID_TOKEN);
+
+        RegisterResponse registerResponse = clientRegistration(redirectUri, responseTypes,
+                SubjectType.PUBLIC, subjectIdentifierAttribute, false);
+
+        String clientId = registerResponse.getClientId();
+
+        List<String> scopes = Arrays.asList("openid", "profile", "address", "email");
+        String state = UUID.randomUUID().toString();
+        String nonce = UUID.randomUUID().toString();
+
+        AuthorizationRequest authorizationRequest = new AuthorizationRequest(responseTypes, clientId, scopes, redirectUri, nonce);
+        authorizationRequest.setState(state);
+
+        AuthorizationResponse authorizationResponse = authenticateResourceOwnerAndGrantAccess(
+                authorizationEndpoint, authorizationRequest, userId, userSecret);
+
+        assertNotNull(authorizationResponse.getLocation());
+        assertNotNull(authorizationResponse.getAccessToken());
+        assertNotNull(authorizationResponse.getState());
+        assertNotNull(authorizationResponse.getScope());
```

Comment:
We can use `io.jans.as.client.client.Asserter#assertAuthorizationResponse(io.jans.as.client.AuthorizationResponse)` to reduce number of asserts which as "standard" (repeat from test to test).